modules/enterprise-management/src/test/java/com/ibm/cloud/platform_services/enterprise_management/v1/EnterpriseManagementIT.java

```diff
@@ -1,433 +1,563 @@
+              .intValue()))
+          .getId();
+      secondExampleAccountGroupId = listAccountGroupsResponseResult
+          .getResources()
+          .get(rnd.nextInt(listAccountGroupsResponseResult
+              .getRowsCount()
+              .intValue()))
+          .getId();
+
+    } catch (ServiceResponseException e) {
+      fail(String.format("Service returned status code %d: %s%nError details: %s", e.getStatusCode(), e.getMessage(),
+          e.getDebuggingInfo()));
+    }
+  }
+
+  @Test
+  public void testListAccountGroupsWithPaging() throws Exception {
+    try {
+      List<AccountGroup> accountGroupsList = new ArrayList<>();
+      String nextDocId = null;
+      Integer limit = 10;
+      do {
+        ListAccountGroupsOptions listAccountGroupsOptions = new ListAccountGroupsOptions.Builder()
+            .enterpriseId(enterpriseId)
+            .nextDocid(nextDocId)
+            .limit(limit)
+            .build();
+
+        // Invoke operation
+        Response<ListAccountGroupsResponse> response = service
+            .listAccountGroups(listAccountGroupsOptions)
+            .execute();
+        // Validate response
         assertNotNull(response);
         assertEquals(response.getStatusCode(), 200);
-        Account responseObj = response.getResult();
-        assertNotNull(responseObj);
-    }
-
-    @Test
-    public void test12_CreateAccountGroupWOptions() throws Throwable {
-        CreateAccountGroupOptions createAccountGroupOptionsModel = new CreateAccountGroupOptions.Builder()
-                .parent(parent).name("testString").primaryContactIamId(ownerIamId).build();
 
-        Response<CreateAccountGroupResponse> response = service.createAccountGroup(createAccountGroupOptionsModel)
-                .execute();
-        assertNotNull(response);
-        assertEquals(response.getStatusCode(), 201);
-        CreateAccountGroupResponse responseObj = response.getResult();
-        assertNotNull(responseObj);
-        accountGroupId = responseObj.getAccountGroupId();
-    }
-
-    @Test
-    public void test13_ListAccountGroupsWOptions() throws Throwable {
-        ListAccountGroupsOptions listAccountGroupsOptions = new ListAccountGroupsOptions.Builder().parent(parent)
-                .enterpriseId(enterpriseId).parentAccountGroupId(accountGroupId).limit(100).build();
-
-        Response<ListAccountGroupsResponse> response = service.listAccountGroups(listAccountGroupsOptions).execute();
+        ListAccountGroupsResponse listAccountGroupsResponseResult = response.getResult();
+
+        assertNotNull(listAccountGroupsResponseResult);
+        assertNotNull(listAccountGroupsResponseResult.getResources());
+        accountGroupsList.addAll(listAccountGroupsResponseResult.getResources());
+
+        nextDocId = getNextDocId(listAccountGroupsResponseResult.getNextUrl());
+      } while (nextDocId != null);
+
+      log(String.format("Received a total of %d account groups.%n", accountGroupsList.size()));
```

Comment:
One additional check you could make here is to walk through `accountGroupsList` and make sure it contains the id's of both of the account groups you created above.